model/project_ref.go

```diff
@@ -133,6 +135,58 @@ type ProjectRef struct {
 	// Filter/view settings
 	ProjectHealthView ProjectHealthView `bson:"project_health_view" json:"project_health_view" yaml:"project_health_view"`
 	ParsleyFilters    []parsley.Filter  `bson:"parsley_filters,omitempty" json:"parsley_filters,omitempty"`
+
+	// GitHubTokenPermissionByRequester contains what permissions each requester has for each token.
+	// By default, requesters have inherit all of the GitHub app permissions. Once a single permission
+	// is specified for a requester, it will only have those permissions.
+	GitHubTokenPermissionByRequester GitHubDynamicTokenPermissions `bson:"github_token_permission_by_requester,omitempty" json:"github_token_permission_by_requester,omitempty" yaml:"github_token_permission_by_requester,omitempty"`
+}
+
+// GitHubDynamicTokenPermissions is used to specify groups of permissions and requesters for
+// GitHub dynamic access tokens.
+type GitHubDynamicTokenPermissions []GitHubDynamicTokenPermission
+
+// GitHubDynamicTokenPermission is used to specify an individiual group of permissions and requesters
+// for GitHub dynamic access tokens.
+type GitHubDynamicTokenPermission struct {
+	// Requesters is a list of the Evergreen requester from globals.go.
+	Requesters []string `bson:"requesters,omitempty" json:"requesters,omitempty" yaml:"requesters,omitempty"`
```

Comment:
One downside to not storing it as a map (meaning requester: permission) is that we can technically add duplicates. We need to add some logic that checks if that requester is already accounted for before adding it.